### PR TITLE
Support transaction mode for mysql

### PR DESCRIPTION
**Pull request by nati**
This commit need to be tested in integraion test
Closes: Issue55

### CODE REVIEW

File: docs/source/schema.rst [hunk 1]
```diff
@@ -531,3 +531,22 @@ Available events
       "delay": "1h"
     }
   }
+
+Custom Isolation Level
+-------------------------
+
+You can specify transaction isolation level for api request.
```
Comment (pedro-r-marques) [inline docs/source/schema.rst]: You can specify the (article) transaction isolation level for api requests (plural).

File: docs/source/schema.rst [hunk 2]
```diff
@@ -531,3 +531,22 @@ Available events
...
+Currently, this is only supported for mysql.
+By default, level for read is repeatable read and the others are
```
Comment (pedro-r-marques) [inline docs/source/schema.rst]: The default setting is "read repeatable" for read operations and "serializable" for operations that modify the database (create, update, delete).

File: schema/schema.go [hunk 3]
```diff
@@ -149,7 +150,15 @@ func NewSchemaFromObj(rawTypeData interface{}) (*Schema, error) {
 	if !ok {
 		return nil, &typeAssertionError{"singular"}
 	}
-
+	isolationLevel, _ := typeData["isolation_level"].(map[string]interface{})
+	if isolationLevel == nil {
+		isolationLevel = map[string]interface{}{
+			ActionRead:   "REPEATABLE READ",
```
Comment (pedro-r-marques) [inline schema/schema.go]: It would be desirable to create constants for the isolation levels.

File: server/resources/resource_management.go [hunk 4]
```diff
@@ -67,7 +67,7 @@ type ExtensionError struct {
 }
 
 //InTransaction executes function in the db transaction and set it to the context
-func InTransaction(context middleware.Context, dataStore db.DB, f func() error) error {
+func InTransaction(context middleware.Context, dataStore db.DB, level string, f func() error) error {
```
Comment (pedro-r-marques) [inline server/resources/resource_management.go]: suggestion:
define "type IsolationLevel string" and use it as argument

File: docs/source/schema.rst (hunk 1 above)
Comment (nati) [inline docs/source/schema.rst]: done

File: docs/source/schema.rst (hunk 2 above)
Comment (nati) [inline docs/source/schema.rst]: done

File: schema/schema.go (hunk 3 above)
Comment (nati) [inline schema/schema.go]: done

File: server/resources/resource_management.go (hunk 4 above)
Comment (nati) [inline server/resources/resource_management.go]: done

**Comment by pedro-r-marques** [comment]
LGTM

**Comment by nati** [comment]
Thanks!

**Pull request merged**